Demo/Iterator/Directory.cpp

```diff
@@ -0,0 +1,11 @@
+#include <iostream>
+#include "Iterator/Directory.h"
+
+int main()
+{
+    Directory dir("some path");
```

Comment:
Your demo won't work if I clone and then build your repository.
Consult with Lilit and Varditer about a (temporary) solution I shared with them.